src/nrtk_explorer/app/export.py

```diff
@@ -0,0 +1,216 @@
+from pathlib import Path
+
+from trame.app import get_server, asynchronous
+from trame.widgets import quasar, html
+from trame.ui.quasar import QLayout
+
+from nrtk_explorer.app.applet import Applet
+from nrtk_explorer.library.dataset import (
+    discover_datasets,
+    dataset_select_options,
+)
+import nrtk_explorer.library.transforms as trans
+from nrtk_explorer.widgets.nrtk_explorer import ExportWidget
+
+
+def recursive_rmdir(path: Path):
+    if not path.is_dir():
+        return
+
+    for item in path.iterdir():
+        if item.is_file():
+            item.unlink()
+        else:
+            recursive_rmdir(item)
+
+    path.rmdir()
+
+
+class ExportApp(Applet):
+    def __init__(self, server):
+        super().__init__(server)
+
+        self.context.setdefault("repository", "")
+        self.state.setdefault("current_dataset", "")
+        self.state.setdefault("repository_datasets", [])
+        self.state.setdefault("export_status", "idle")
+        self.state.setdefault("export_progress", 0)
+
+        self.server.controller.add("on_server_ready")(self.on_server_ready)
+
+        self._ui = None
+
+    def on_server_ready(self, *args, **kwargs):
+        # Bind instance methods to state change
+        pass
+
+    def on_export_clicked(self, event):
+        self.start_export(event["name"], event["full"])
+
+    def start_export(self, name, full):
+        if self._exporting_dataset():
+            return
```

Comment:
User could have stale images in the export if they change transforms, then click export again?